db/db.go

```diff
@@ -80,62 +69,45 @@ func Open(dbPath string) (*DB, error) {
 	openMu.Lock()
 	defer openMu.Unlock()
 
-	dbc, err := sql.Open("sqlite3_with_hook", dbPath)
+	os.MkdirAll(filepath.Dir(dbPath), 0755)
+	d := sqlite3.SQLiteDriver{}
+	dbc, err := d.Open(dbPath)
+
 	if err != nil {
 		return nil, err
 	}
 
-	// Ensure a connection is established before going further. Ignore error
-	// because database may not exist yet.
-	_ = dbc.Ping()
-
 	return &DB{
-		conn:        dbc,
-		sqlite3conn: sqlite3conn,
+		sqlite3conn: dbc.(*sqlite3.SQLiteConn),
 		path:        dbPath,
 	}, nil
 }
 
 // OpenWithConfiguration an existing database, creating it if it does not exist.
 func OpenWithConfiguration(dbPath string, conf *Config) (*DB, error) {
-	openMu.Lock()
-	defer openMu.Unlock()
-
-	dbc, err := sql.Open("sqlite3_with_hook", conf.FQDSN(dbPath))
-	if err != nil {
-		return nil, err
-	}
-
-	// Ensure a connection is established before going further. Ignore error
-	// because database may not exist yet.
-	_ = dbc.Ping()
-
-	return &DB{
-		conn:        dbc,
-		sqlite3conn: sqlite3conn,
-		path:        conf.FQDSN(dbPath),
-	}, nil
+	return Open(conf.FQDSN(dbPath))
 }
 
 // Close closes the underlying database connection.
 func (db *DB) Close() error {
-	return db.conn.Close()
+	return db.sqlite3conn.Close()
 }
 
 // Execute executes queries that modify the database.
 func (db *DB) Execute(queries []string, tx, xTime bool) ([]*Result, error) {
 	type Execer interface {
-		Exec(query string, args ...interface{}) (sql.Result, error)
+		Exec(query string, args []driver.Value) (driver.Result, error)
 	}
 
 	var allResults []*Result
-	err := func() error {
+	err := func() (err error) {
```

Comment:
If I don't name the error, then I get this:
```
src/github.com/otoolep/rqlite/db/db.go:126: undefined: err
src/github.com/otoolep/rqlite/db/db.go:127: undefined: err
src/github.com/otoolep/rqlite/db/db.go:128: undefined: err
```
If use `:=` to try and fix the above problem, I get this:
```
src/github.com/otoolep/rqlite/db/db.go:126: t declared and not used
```
I'm open to suggestions if there is a better way to achieve the intended result.